lib/spack/spack/cmd/checkout.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import llnl.util.filesystem as fs
+import llnl.util.tty as tty
+
+import spack.environment as ev
+import spack.paths
+import spack.repo
+from spack.util.executable import which
+
+_SPACK_UPSTREAM = 'https://github.com/spack/spack'
+
+description = "update the spack prefix or subcomponent to a git ref"
+section = "admin"
+level = "long"
+
+git = None
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        '-r', '--remote', action='store', default='origin',
+        help="name of the git remote from which to fetch")
+    subparser.add_argument(
+        '--url', action='store', default=None,
+        help="url to use if the remote does not already exist")
+
+    subrepo_subparser = subparser.add_mutually_exclusive_group()
+    subrepo_subparser.add_argument(
+        '--env', action='store', default=None,
+        help="checkout an environment instead of the Spack source")
```

Comment:
Where does the environment go?  This one's confusing to me -- what's it supposed to do?  Check out a named env within spack?